pulsar-broker/src/test/java/org/apache/pulsar/broker/service/ReplicatorTest.java

```diff
@@ -1046,7 +1045,7 @@ public void testResumptionAfterBacklogRelaxed() throws Exception {
      *
      * @throws Exception
      */
-    @Test(timeOut = 15000)
+    @Test(timeOut = 30000)
     public void testCloseReplicatorStartProducer() throws Exception {
```

Comment:
Start a channel to discuss the following topic

@poorbarcode left a comment

> Multiple threads attempt to cancel pending read requests concurrently
> Cursors are closed while read operations are pending
> OpReadEntry instances are recycled and reused incorrectly

Could you describe the details of how the `OpReadEntry` will be reused

---

@lhotari  left a comment

Since OpReadEntry instances are recycled, they are reused. It's not sufficient do use compareAndSet. There's a need to have some unique identifier on the instance to do the CAS operation comparison. It's possible to find code paths where this could happen and new problems could be introduced unless we have a proper solution to for CAS operations.